src/Compilers/Core/Portable/Compilation/Compilation.cs

```diff
@@ -1480,7 +1480,12 @@ internal bool IsRealSigned
             DiagnosticBag diagnostics,
             CancellationToken cancellationToken);
 
-        internal abstract void ReportUnusedImports(
+        /// <summary>
+        /// Reports all unused inports/usings so far (and thus it must be called as a last step of Emit)
+        /// Also signals the event queue, if any, that we are done compiling.
+        /// There should not be more compiling actions after this step.
```

Comment:
Right. Will add a note and point to the possible bug (or feature?) that some diagnostics may be cut off.